docs/architecture.md

```diff
@@ -307,3 +290,62 @@ for more information.
+
+## Features
+
+### Antrea Network Policy
+Besides Kubernetes NetworkPolicy, Antrea supports two extra types of
+Network Policies available as CRDs - Antrea Namespaced NetworkPolicy and
+ClusterNetworkPolicy. The former is scoped to a specific Namespace, while the
+latter is scoped to the whole cluster. These two types of Network Policies
+extended Kubernetes NetworkPolicy with advanced features including: policy
+priority, tiering, deny action, external entity, and policy statistics. For more
+information about Antrea network policies, refer to the [Antrea Network Policy document](antrea-network-policy.md).
+
+The same as Kubernetes NetworkPolicies, Antrea Controller computes Antrea
+NetworkPolicies and ClusterNetworkPolicies to internal NetworkPolicy,
+AddressGroup and AppliedToGroup objects, and disseminates them to Antrea
+Agents. Antrea Agents create OVS flows to enforce the NetworkPolicies applied
+to the local Pods on their Nodes.
+
+### IPsec encryption
+Antrea supports encrypting GRE tunnel traffic with IPsec ESP. The IPsec
+implementation leverages [OVS IPsec](http://docs.openvswitch.org/en/latest/tutorials/ipsec)
+and leverages [strongSwan](https://www.strongswan.org) as the IKE daemon.
+
+To enable IPsec, an extra container -`antrea-ovs-ipsec` - must be added to the
+Antrea Agent DaemonSet, which runs the `ovs-monitor-ipsec` and strongSwan
+daemons. Antrea now supports only using pre-shared key (PSK) for IKE
+authentication, and the PSK string must be passed to Antrea Agent using an
+environment variable - `ANTREA_IPSEC_PSK`. The PSK string can be specified in
+the [Antrea IPsec deployment yaml](/build/yamls/antrea-ipsec.yml), which creates
+a Kubernetes Secret to save the PSK value and populates it to the
+`ANTREA_IPSEC_PSK` environment variable of the Antrea Agent container.
+
+When IPsec is enabled, Antrea Agent will create a separate GRE tunnel port on
+the OVS bridge for each remote Node, and write the PSK string and the remote
+Node IP address to two OVS interface options of the tunnel interface. Then
+`ovs-monitor-ipsec` can detect the tunnel and create IPsec Security Policies
+with PSK for the remote Node, and strongSwan can create the IPsec Security
+Associations based on the Security Policies. These additional tunnel ports are
+not used to send traffic to a remote Node - the tunnel traffic is still output
+to the default tunnel port (`antrea-tun0`) with OVS flow based tunneling.
+However, the traffic from a remote Node will be received from the Node's IPsec
+tunnel port.
+
+### Network flow visibility
+Antrea supports exporting network flow information with Kubernetes contexts
+using IPFix. The exported network flows can be visualized using Elastic Stack
+and Kibana dashboards. For more information, refer to the [network flow
+visibility document](network-flow-visibility.md).
+
+### Windows Node
+On a Windows Node, Antrea acts very like it does on a Linux Node. Antrea Agent
```

Comment:
s/very like it does/very much like it does

Reply:
Done